### DISCUSSION TITLE
run time visualisation

### DISCUSSION THREAD

**Discussion by mj-di** [Mair/esp32-course#96] [Course topic requests]
Hi all

Are any resources availabe for the hobbyist to visualise the run time operation of the RTOS. Something better than jsut guessing priorities etc and test observe external performance in terms of expected responses performed by different threads - then tweak appropriately. Some disciplined approach / reporting of metrics would be useful. For the professional developer tools like Tracealyzer [https://percepio.com/tracealyzer/](url) are available, is there a free similar tool available?

Thanks for any comments / tips

mjd

**Comment by Mair** [comment]
Hello @mj-di,
I've used Segger's [system view](https://www.segger.com/products/development-tools/systemview/) before. Its available for free when used non-commercially. 
The way it works is you turn on tracing in the menuconfig and run the app. This produces a trace "dump" Then you can load the "dump" into the tool and analyze it. Instructions on how to do that [here](https://www.segger.com/products/development-tools/systemview/)

**Reply by mj-di** [reply]
thank you Mair, will take a look
